flink-python/pyflink/datastream/data_stream.py

```diff
@@ -0,0 +1,162 @@
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+from pyflink.common import typeinfo, ExecutionConfig
+from pyflink.common.typeinfo import TypeInformation
+
+
+class DataStream(object):
+    """
+    A DataStream represents a stream of elements of the same type. A DataStream can be transformed
+    into another DataStream by applying a transformation as for example:
+    ::
+        >>> DataStream.map(MapFunctionImpl())
+        >>> DataStream.filter(FilterFunctionImpl())
+    """
+
+    def __init__(self, j_data_stream):
+        self._j_data_stream = j_data_stream
+
+    def get_name(self) -> str:
+        """
+        Gets the name of the current data stream. This name is used by the visualization and logging
+        during runtime.
+
+        :return: Name of the stream.
+        """
+        return self._j_data_stream.getName()
+
+    def name(self, name: str):
```

Comment:
Should also add tests to verify these methods.

Reply:
Yes, I will add the corresponding tests to verify them.